src/sync.ts

```diff
@@ -554,134 +622,136 @@ export class Sync {
+            const parentPath = this.getParentPath(nodePath);
+            const parentNode = changedNodes[parentPath];
+            const itemName = nodePath.substring(path.length);
+            if (parentNode && parentNode.local) {
+              delete parentNode.local.itemsMap[itemName];
+
+              if (equal(parentNode.local.itemsMap, parentNode.common.itemsMap)) {
+                delete parentNode.local;
               }
             }
           }
         }
       }
-
-      return this.deleteRemoteTrees(Object.keys(recurse), changedNodes)
-        .then(changedObjs2 => {
-          return this.rs.local.setNodes(this.flush(changedObjs2));
-        });
-    });
-  }
-
-  public async deleteRemoteTrees (paths: Array<string>, changedNodes: RSNodes): Promise<RSNodes> {
-    if (paths.length === 0) {
-      return Promise.resolve(changedNodes);
     }
 
-    return this.rs.local.getNodes(paths).then(async (nodes: RSNodes) => {
-      const subPaths = {};
+    const changedNodes2 = await this.markRemoteDeletions(Object.keys(recurse), changedNodes);
+    if (changedNodes2) {
+      await this.rs.local.setNodes(this.flush(changedNodes2));
+    }
+  }
 
-      function collectSubPaths (folder, path: string): void {
-        if (folder && folder.itemsMap) {
-          for (const itemName in folder.itemsMap) {
-            subPaths[path+itemName] = true;
-          }
+  /**
+   * Recursively process paths to mark documents as remotely deleted
+   * where applicable
+   */
+  async markRemoteDeletions (paths: string[], changedNodes: RSNodes): Promise<RSNodes | void> {
+    if (paths.length === 0) { return changedNodes; }
+
+    const nodes = await this.rs.local.getNodes(paths);
+    const subPaths = {};
+
+    function collectSubPaths (folder: RSItem, path: string): void {
+      if (folder && folder.itemsMap) {
+        for (const itemName in folder.itemsMap) {
+          subPaths[path+itemName] = true;
         }
       }
+    }
 
-      for (const path in nodes) {
-        const node = nodes[path];
-
-        // TODO Why check for the node here? I don't think this check ever applies
-        if (!node) { continue; }
+    for (const path in nodes) {
+      const node = nodes[path];
+      if (!node) { continue; }
 
-        if (isFolder(path)) {
-          collectSubPaths(node.common, path);
-          collectSubPaths(node.local, path);
-        } else {
-          if (node.common && typeof(node.common.body) !== undefined) {
-            changedNodes[path] = deepClone(node);
-            changedNodes[path].remote = {
-              body:      false,
-              timestamp: this.now()
-            };
-            changedNodes[path] = this.autoMerge(changedNodes[path]);
-          }
+      if (isFolder(path)) {
+        collectSubPaths(node.common, path);
+        collectSubPaths(node.local, path);
+      } else {
+        if (node.common && typeof(node.common.body) !== undefined) {
```

Comment:
`typeof` always returns a string.

```suggestion
        if (node.common && typeof(node.common.body) !== 'undefined') {
```